atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -240,7 +240,8 @@ public static SerializableTransactionManager create(
         kvs = AtlasDbMetrics.instrument(KeyValueService.class, kvs, MetricRegistry.name(KeyValueService.class));
         kvs = ValidatingQueryRewritingKeyValueService.create(kvs);
 
-        TransactionManagersInitializer.createInitialTables(kvs, schemas, config.initializeAsync());
+        TransactionManagersInitializer initializer = TransactionManagersInitializer.createInitialTables(kvs, schemas,
```

Comment:
Any reason we don't just have `SerializableTransactionManager` extend `AsyncInitializer` and put this logic in there?

(it seems like a bit of a snowflake where its initialization logic is in some random external class, rather than an inner class like the rest)

Reply:
Not that I'm advocating making a big refactor here, this isn't that bad, just wondering if this was an intentional design choice

Reply:
I think it's because we require certain setup steps to be performed before we try this (creating the KVS and Lock and Timestamp services, for instance).

Also, the startup ordering PR deliberately tried to keep things the same for the synchronously initialised world. `TransactionManagers.create` does a whole bunch of things, and the order of the things was maintained.

Moving this logic to a proper TransactionManager initialiser shouldn't be done automatically, as it would alter this order, and could have side-effects - for instance, we would be creating various services before `createInitialTables` rather than after. Is this safe?